csharp/FileManager/Manager.cs

```diff
@@ -0,0 +1,138 @@
+            var sequenceWalker = new RightSequenceWalker<TLinkAddress>(Links, new DefaultStack<TLinkAddress>(), unicodeSymbolCriterionMatcher.IsMatched);
+            _stringToUnicodeSequenceConverter = new CachingConverterDecorator<string, TLinkAddress>(new StringToUnicodeSequenceConverter<TLinkAddress>(Links, charToUnicodeSymbolConverter, balancedVariantConverter, _unicodeSequenceMarker));
+            _unicodeSequenceToStringConverter = new CachingConverterDecorator<TLinkAddress, string>(new UnicodeSequenceToStringConverter<TLinkAddress>(Links, unicodeSequenceCriterionMatcher, sequenceWalker, unicodeSymbolToCharConverter));
+        }
+
+        private TLinkAddress GetFileLink(string name)
+        {
+            var nameLink = _stringToUnicodeSequenceConverter.Convert(name);
+            var any = Links.Constants.Any;
+            var query = new Link<UInt64>(index: any, source: nameLink, target: any);
+            var list = new List<IList<TLinkAddress>>();
+            var listFiller = new ListFiller<IList<TLinkAddress>, TLinkAddress>(list, Links.Constants.Continue);
+            Links.Each(listFiller.AddAndReturnConstant, query);
+            if (list.Count != 0)
+            {
+                return list.First()[Links.Constants.TargetPart];
+            }
+            else
+            {
+                return 0;
+            }
+        }
+
+        public string PutFile(string addres)
+        {
+            if(GetFileLink(addres) != 0)
+            {
+               return _unicodeSequenceToStringConverter.Convert(GetFileLink(addres));
+            }
+            else
+            {
+                return "File does not exists";
+            }
+        }
+
+        public TLinkAddress AddFile(string name, string content)
+        {
+            var source = _stringToUnicodeSequenceConverter.Convert(name);
+            var target = _stringToUnicodeSequenceConverter.Convert(content);
+            return Links.GetOrCreate(source, target);
+        }
+
+        public void Delete(TLinkAddress link) => Links.Delete(link);
+
+        public void Delete(string addres)
+        {
+            if (GetFileLink(addres) != 0)
+            {
+                Links.Delete(GetFileLink(addres));
+            }
+        }
+
+        public string GetAllLinks()
+        {
+            var any = Links.Constants.Any;
+            string links = "";
+            var query = new Link<UInt64>(index: any, source: any, target: any);
+            Links.Each((link) => {
+                links += Links.Format(link) + "\n";
+                return Links.Constants.Continue;
+            }, query);
+            return links;
+        }
+
+        public bool LinkExist(string addres)
+        {
+            if (GetFileLink(addres) != 0)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }

```

Comment:
No need to write so much code for this.

Your function can be expressed as equivalent single line: `return GetFileLink(addres) != 0;`.